### ISSUE TITLE
Remove twisted.mail.protocols.SSLContextFactory

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8591]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8591|
|Type|enhancement|
|Created|2016-07-12 11:39:45Z|
|Branch|https://github.com/twisted/twisted/tree/8591-t.m.protocols.SSLContextFactory-removal|

It is about time :)

<details><summary>Searchable metadata</summary>

```
trac-id__8591 8591
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__8591_t_m_protocols_SSLContextFactory_removal 8591-t.m.protocols.SSLContextFactory-removal
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__None None
time__1468323585528043 1468323585528043
changetime__1469022578457621 1469022578457621
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban removed owner|
|-|-|

Please check that changes are ok. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set owner to @adiroiban|
|-|-|

Looks OK.
Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset b78ea5407961b408b60f079f074f0a253f3e070c
```
#!CommitTicketReference repository="" revision="b78ea5407961b408b60f079f074f0a253f3e070c"
Merge 8591-t.m.protocols.SSLContextFactory-removal: Remove twisted.mail.protocols.SSLContextFactory

Author: adiroiban
Reviewer: rodrigc
Fixes: #8591
```
